### PR TITLE
vembrane

### PR DESCRIPTION
**Pull request by christopher-schroeder**
add vembrane

### CODE REVIEW

File: bio/vembrane/environment.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+channels:
+  - bioconda
+  - conda-forge
+  - default
+dependencies:
+  - vembrane =0.3.2
```
Comment (dlaehnemann) [inline bio/vembrane/environment.yaml:6]: Update to `0.4.0`? :D

File: bio/vembrane/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+name: vembrane
+description: Vembrane allows to simultaneously filter variants based on any INFO field, CHROM, POS, REF, ALT, QUAL, and the annotation field ANN. When filtering based on ANN, annotation entries are filtered first. If no annotation entry remains, the entire variant is deleted.
```
Comment (dlaehnemann) [inline bio/vembrane/meta.yaml:2]: Maybe link to the docs / the repo?

File: bio/vembrane/wrapper.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+from snakemake.shell import shell
+
+log = snakemake.log_fmt_shell(stdout=False, stderr=True)
+
+extra = snakemake.params.get("extra", "")
+
+shell(
+    "vembrane"  # Tool and its subcommand
+    " {extra}"  # Extra parameters
+    ' "{snakemake.params.expression}"'
```
Comment (dlaehnemann) [inline bio/vembrane/wrapper.py:17]: Nice, I didn't know you could mix single and double quotes within a multi-line string!

File: bio/vembrane/environment.yaml (hunk 1 above)
Comment (christopher-schroeder) [reply bio/vembrane/environment.yaml]: Till was too fast! :D

File: bio/vembrane/meta.yaml (hunk 2 above)
Comment (christopher-schroeder) [reply bio/vembrane/meta.yaml]: I linked to the repo, good idea

File: bio/vembrane/wrapper.py (hunk 3 above)
Comment (christopher-schroeder) [reply bio/vembrane/wrapper.py]: Now that you mentioned, I cannot unsee it! I had to change it :)

**Comment by dlaehnemann** [comment]
Hmm, with version `0.4.0` it now throws a KeyError on the `INFO` field. Sounds like it is trying to parse header lines for the `INFO` fields and none are there, so maybe `vembrane` needs to cover the corner case that no `INFO` fields are used:

```
Traceback (most recent call last):
  File "/home/runner/work/snakemake-wrappers/snakemake-wrappers/bio/vembrane/test/.snakemake/conda/51a99868/bin/vembrane", line 10, in <module>
    sys.exit(main())
  File "/home/runner/work/snakemake-wrappers/snakemake-wrappers/bio/vembrane/test/.snakemake/conda/51a99868/lib/python3.7/site-packages/vembrane/cli.py", line 90, in main
    first_record = list(islice(records, 1))
  File "/home/runner/work/snakemake-wrappers/snakemake-wrappers/bio/vembrane/test/.snakemake/conda/51a99868/lib/python3.7/site-packages/vembrane/cli.py", line 113, in filter_vcf
    env = Environment(expression, ann_key, vcf.header)
  File "/home/runner/work/snakemake-wrappers/snakemake-wrappers/bio/vembrane/test/.snakemake/conda/51a99868/lib/python3.7/site-packages/vembrane/representations.py", line 220, in __init__
    self._header_info_fields = self._numbers["INFO"]
KeyError: 'INFO'
```

**Comment by jafors** [comment]
> Hmm, with version `0.4.0` it now throws a KeyError on the `INFO` field. Sounds like it is trying to parse header lines for the `INFO` fields and none are there, so maybe `vembrane` needs to cover the corner case that no `INFO` fields are used:

I think this is fixed in the recent release. So, we just have to wait for the update. :)

**Comment by christopher-schroeder** [comment]
Yes, till fixed that in 0.4.1. Just waiting for the bioconda bot.
